metadata/src/main/java/org/apache/kafka/controller/ClusterControlManager.java

```diff
@@ -297,12 +310,17 @@ private ClusterControlManager(
         this.controllerRegistrations = new TimelineHashMap<>(snapshotRegistry, 0);
         this.directoryToBroker = new TimelineHashMap<>(snapshotRegistry, 0);
         this.brokerUncleanShutdownHandler = brokerUncleanShutdownHandler;
+        this.nodeId = nodeId;
     }
 
     ReplicaPlacer replicaPlacer() {
         return replicaPlacer;
     }
 
+    public int nodeId() {
+        return nodeId;
+    }
```

Comment:
What's this for? I don't see it used anywhere.

Reply:
This is related to this [comment](https://github.com/apache/kafka/pull/16866#discussion_r1716439359), which is because we didn't trigger unclean leader election when topic config altered.

Reply:
I'm going to remove this because we don't need it at this point